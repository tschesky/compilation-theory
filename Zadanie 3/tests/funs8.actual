Visiting program
Argument a already defined. Line: 4
Type checking finished
