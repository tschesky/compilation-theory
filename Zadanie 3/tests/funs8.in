
int a = 0, b = 0, c = 0;

int bad1(int a, int a) {
    int c = 0;

    while( a != 0 ) {
        c = a;
        a = b % a;
        b = c;
    }
    return b;
}

int bad2(int a, int b) {
    int a = 0;

    while( a != 0 ) {
        c = a;
        a = b % a;
        b = c;
    }
    return b;
}

print bad1(125,100);
print bad2(125,100);

