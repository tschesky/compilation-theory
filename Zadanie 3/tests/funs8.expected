Error: Variable 'a' already declared: line 4
Error: Variable 'a' already declared: line 16
